File: app/src/main/java/com/theroyalsoft/telefarmer/utils/PreOfferAnswerSocket.java
package com.theroyalsoft.telefarmer.utils;

import com.theroyalsoft.telefarmer.helper.AppKey;

/**
 * Created by Pritom Dutta on 5/8/23.
 */
public class PreOfferAnswerSocket {
    private String preOfferAnswer;
    private String fromId;
    private String toId;
    private String toDeviceId;
    private String fromDeviceId;
    private String channel;


    public PreOfferAnswerSocket(String preOfferAnswer, String fromId, String toId) {
        this.preOfferAnswer = preOfferAnswer;
        this.fromId = fromId;
        this.toId = toId;
        this.fromDeviceId = DeviceIDUtil.getDeviceID();
        this.toDeviceId = SocketKey.getReceiverDeviceId(); //Receiver DevicesId
        this.channel = AppKey.CHANNEL;
    }

    public String getFromDeviceId() {
        return NullRemoveUtil.getNotNull(fromDeviceId);
    }

    public String getPreOfferAnswer() {
        return NullRemoveUtil.getNotNull(preOfferAnswer);
    }
}

File: model/src/main/java/bio/medico/patient/model/socket/PreOfferAnswerSocket.kt
package bio.medico.patient.model.socket

import bio.medico.patient.model.NullRemoveUtil

data class PreOfferAnswerSocket(
    private val preOfferAnswer: String,
    private val fromId: String,
    private val toId: String,
    private val fromDeviceId: String,
    private val toDeviceId: String,
    private val channel: String
) {
    fun getFromDeviceId(): String {
        return NullRemoveUtil.getNotNull(fromDeviceId)
    }

    fun getPreOfferAnswer(): String {
        return NullRemoveUtil.getNotNull(preOfferAnswer)
    }
}